DiscordBot.DataAccess/EventsSheetsService.cs

```diff
@@ -217,7 +228,11 @@ public async Task<IEnumerable<DiscordEvent>> ListEventsAsync()
                         (string) row[NameColumn.Index],
                         (string) row[DescriptionColumn.Index],
                         int.Parse((string) row[KeyColumn.Index]),
-                        (DateTime) row[TimeColumn.Index]
+                        (DateTime) row[TimeColumn.Index],
+                        (string) row[TimeZoneColumn.Index],
+                        row.Count >= MessageIdColumn.Index + 1 && (string) row[MessageIdColumn.Index] != ""
```

Comment:
Could you explain this logic please? 🙂

Reply:
Yeah this a bit complicated and probably should have a comment. The `messageId` might be `null`, and this conditional is deciding whether the value is `null`, or an actual ID, and parsing or returning `null` appropriately. 

To explain the conditional more specifically: the get requests trim trailing empty cells (so the length of the `row` array depends on the values in that row), so if the last cell containing this value is empty (ie the `messageId` is `null`), then `row[MessageIdColumn.Index]` will be out of bounds, so `null` should just returned instead. The second case checks for the cell being in the `row` array, but containing an empty string. This will also fail to parse, and corresponds to a `null` value. This will not happen in the current sheet layout, but if there is data in the cell to the right of the row then this cell will be `""` rather than being trimmed from the end of the array. I have explained that in a way that that makes sense? 🙂

Reply:
Ah okay, I think I see. It might help to extract this to a function, e.g. `ParseMessageId`, where you create suitably name variables for each part of this condition. That should hopefully make it easier to understand. What do you think? 🙂

Reply:
Yes I agree, done in 483918a 🙂